### PR TITLE
feat(nacos): support reuse nacos client

### PR DESCRIPTION
**Pull request by whalecold**
#### What type of PR is this?

feature
#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [x] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en:
zh(optional): 

#### (Optional) Which issue(s) this PR fixes:

https://github.com/kitex-contrib/config-nacos/issues/9

### CODE REVIEW

File: nacos/uid.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nacos
+
+import "go.uber.org/atomic"
+
+var uniqueID atomic.Int64
```
Comment (felix021) [inline nacos/uid.go:19]: `atomic.Int64` is not available until go1.19, so you might need to implement one yourself

Comment (whalecold) [reply nacos/uid.go:19]: I use "go.uber.org/atomic" other than "sync/atomic"~

File: nacos/nacos.go [hunk 2]
```diff
@@ -174,19 +196,77 @@ func (c *client) configParam(cpc *ConfigParamConfig, t *template.Template) (vo.C
...
+		err := c.ncli.ListenConfig(vo.ConfigParam{
+			DataId:   param.DataId,
+			Group:    param.Group,
+			Content:  param.Content,
+			DatumId:  param.DatumId,
+			Type:     param.Type,
+			OnChange: c.onChange,
+		})
+		if err != nil {
+			panic(err)
```
Comment (felix021) [inline nacos/nacos.go:255]: Better add a comment

> `nacosClient.ListenConfig` performs only local connection and fails only when the input params are invalid

File: nacos/nacos.go [hunk 3]
```diff
@@ -174,19 +196,77 @@ func (c *client) configParam(cpc *ConfigParamConfig, t *template.Template) (vo.C
...
-		klog.Debugf("[nacos] config %s updated, namespace %s group %s dataId %s data %s",
-			param.DataId, namespace, group, dataId, data)
+		klog.Debugf("[nacos] uniqueID %d config %s updated, namespace %s group %s dataId %s data %s",
+			uniqueID, param.DataId, namespace, group, dataId, data)
 		callback(data, c.parser)
 	}
+
 	data, err := c.ncli.GetConfig(param)
 	// the nacos client has handled the not exist error.
 	if err != nil {
```
Comment (felix021) [inline nacos/nacos.go:272]: This panic might be troublesome. What if it's a temporary network failure?

File: nacos/nacos.go [hunk 4]
```diff
@@ -174,29 +196,87 @@ func (c *client) configParam(cpc *ConfigParamConfig, t *template.Template) (vo.C
...
 	}
+
+	// NOTE: does not ensure that GetConfig succeeds, the govern policy may not be correct if it fails here.
 	data, err := c.ncli.GetConfig(param)
-	// the nacos client has handled the not exist error.
 	if err != nil {
-		panic(err)
+		// If the initial connection fails and the reconnection is successful, the callback handler can also be invoked.
+		// Ignore the error here and print the error info.
+		klog.Errorf("get config %v from nacos failed %v", param, err)
```
Comment (felix021) [inline nacos/nacos.go:276]: Better to be a warning here.

Comment (whalecold) [reply nacos/nacos.go]: Done

**Review by felix021 — APPROVED** [review]

**Pull request merged**